src/cheri_cap_common.sail

```diff
@@ -629,9 +598,9 @@ function getCapLength(c) : Capability -> CapLen =
  */
 val inCapBounds  : (Capability, CapAddrBits, CapLen) -> bool
 function inCapBounds (cap, addr, size) = {
-  let (base, top) = getCapBounds(cap);
+  let (base, length) = getCapBounds(cap);
   let a = unsigned(addr);
-  (a >= base) & ((a + size) <= top)
+  (a >= base) & (size <= length)
```

Comment:
This is incorrect. Suppose a capability has base=0, length=10 and we call this function with addr=10, size=1 this function will return true but it should return false.

Reply:
Oops, I got confused between base and a.

Reply:
Fixed